examples/knative/powershell/kn-ps-harbor-slack/README.md

```diff
@@ -0,0 +1,112 @@
+
+# Step 1 - Build
+
+
+Create the container image locally to test your function logic.
+
+```
+export TAG=<version>
+docker build -t <docker-username>/kn-ps-harbor-slack:${TAG} .
+```
+
+# Step 2 - Test
+
+Verify the container image works by executing it locally.
+
+Change into the `test` directory
+```console
+cd test
+```
+
+Update the following variable names within the `docker-test-env-variable` file
+
+* SLACK_WEBHOOK_URL - Slack webhook URL
+* SLACK_MESSAGE_PRETEXT - Text displayed for Slack notification
+
+Start the container image by running the following command:
+
+```console
+docker run -e FUNCTION_DEBUG=true -e PORT=8080 --env-file docker-test-env-variable -it --rm -p 8080:8080 <docker-username>/kn-ps-harbor-slack:${TAG}
+```
+
+In a separate terminal, run either `send-cloudevent-test.ps1` (PowerShell Script) or `send-cloudevent-test.sh` (Bash Script) to simulate a CloudEvent payload being sent to the local container image
+
+```console
+Testing Function ...
+See docker container console for output
+
+# Output from docker container console
+Detected change to subject-123 ...
+Sending Webhook payload to Slack ...
+StatusCode        : 200                                                                                                                                                                                                            StatusDescription : OK                                                                                                                                                                                                             Content           : ok
+RawContent        : HTTP/1.1 200 OK
+                    Date: Tue, 16 Feb 2021 21:56:16 GMT
+                    Server: Apache
+                    Strict-Transport-Security: max-age=31536000; includeSubDomains; preload
+                    x-slack-backend: r
+                    referrer-policy: no-referrer
+                    Vary: Accept-…
+Headers           : {[Date, System.String[]], [Server, System.String[]], [Strict-Transport-Security, System.String[]], [x-slack-backend, System.String[]]…}
+Images            : {}
+InputFields       : {}
+Links             : {}
+RawContentLength  : 2
+RelationLink      : {}
+```
+
+# Step 3 - Deploy
+
+> **Note:** The following steps assume a working Knative environment using the
+`default` Rabbit `broker`. The Knative `service` and `trigger` will be installed in the
+`vmware-functions` Kubernetes namespace, assuming that the `broker` is also available there.
+>
+> **Note:** Also, in order to receive incoming Harbor events and to ultimately invoke the Harbor-Slack-Function, it's necessary to have the [kn-go-harbor-webhook function example](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/development/examples/knative/go/kn-go-harbor-webhook) running properly first.
+
+Push your container image to an accessible registry such as Docker once you're done developing and testing your function logic.
```

Comment:
This is not needed when using the provided function.yaml/image

Reply:
@embano1 Make sense. This would include line 69-71 as well, right?

Reply:
Done